scripts/dialog_xml2json.py

```diff
@@ -230,6 +230,22 @@ def removeAllComments(tree):
         p = c.getparent()
         p.remove(c)
 
+def removeOutOfScopeNodes(tree):
+    scopedNodes = tree.xpath('//*[@scope]')
+    for scopedNode in scopedNodes:
+        if not inScope(scopedNode):
+            p = scopedNode.getparent()
+            p.remove(scopedNode)
+
+def inScope(node):
+    if not hasattr(config, 'common_scope'):
+        return False # no scope specified -> remove all scoped nodes
+    scope = getattr(config, 'common_scope')
+    if scope in node.get('scope'):
```

Comment:
I would at least add comment for this or a TODO, because one could think that you have list of scopes and then you try to find your scope is in the list.... but the code actually do that you have string and try to find a substring... and e.g. if there is value 'abc, efg' in scope in node and you provide scope 'ab', 'c, e', ... it works but it should not... but I did not run it so maybe I am wrong

Reply:
The intent was to allow subscopes, but you are right, 'c, e' should not match. I've changed it to exact match.

Reply:
Maybe I was one step ahead, because I counted with more than one scope defined in tag. But I still think that exact match is something that we want to have implemented now and sub-scopes or multi-scopes do in further steps.